example/lib/pages/interactive_test_page.dart

```diff
@@ -166,4 +164,49 @@ class _InteractiveTestPageState extends State<InteractiveTestPage> {
       ),
     );
   }
+
+  String _eventName(MapEvent? event) {
+    switch (event) {
+      case MapEventTap():
+        return 'MapEventTap';
+      case MapEventSecondaryTap():
+        return 'MapEventSecondaryTap';
+      case MapEventLongPress():
+        return 'MapEventLongPress';
+      case MapEventMove():
+        return 'MapEventMove';
+      case MapEventMoveStart():
+        return 'MapEventMoveStart';
+      case MapEventMoveEnd():
+        return 'MapEventMoveEnd';
+      case MapEventFlingAnimation():
+        return 'MapEventFlingAnimation';
+      case MapEventFlingAnimationNotStarted():
+        return 'MapEventFlingAnimationNotStarted';
+      case MapEventFlingAnimationStart():
+        return 'MapEventFlingAnimationStart';
+      case MapEventFlingAnimationEnd():
+        return 'MapEventFlingAnimationEnd';
+      case MapEventDoubleTapZoom():
+        return 'MapEventDoubleTapZoom';
+      case MapEventScrollWheelZoom():
+        return 'MapEventScrollWheelZoom';
+      case MapEventDoubleTapZoomStart():
+        return 'MapEventDoubleTapZoomStart';
+      case MapEventDoubleTapZoomEnd():
+        return 'MapEventDoubleTapZoomEnd';
+      case MapEventRotate():
+        return 'MapEventRotate';
+      case MapEventRotateStart():
+        return 'MapEventRotateStart';
+      case MapEventRotateEnd():
+        return 'MapEventRotateEnd';
+      case MapEventNonRotatedSizeChange():
+        return 'MapEventNonRotatedSizeChange';
+      case null:
+        return 'null';
+      default:
+        return 'Unknown';
+    }
+  }
```

Comment:
Aware that this is within the example, but would it be better to have each `MapEvent` class define it's friendly name within itself, rather than this massive switch statement?

Reply:
Yeah I hesitated on that too, I ended up putting it in the example because I couldn't imagine why a user would want to use friendly names for events. If they need to match on an event type they can use a case statement. Do you think there is a use for them?